patches/net/minecraft/world/level/block/entity/BlockEntity.java.patch

```diff
@@ -18,6 +20,15 @@
  
      public BlockEntity(BlockEntityType<?> p_155228_, BlockPos p_155229_, BlockState p_155230_) {
          this.type = p_155228_;
+@@ -51,7 +_,7 @@
+     }
+ 
+     public boolean isValidBlockState(BlockState p_353131_) {
+-        return this.type.isValid(p_353131_);
```

Comment:
Maybe add a `// TODO Remove in next breaking phase`? It's just here to unbreak old mods that were made before the vanilla change and is not needed for new mods that can override this method themselves.

Reply:
It shouldn't be a requirement for mods to override both this and `getType`, IMO.